src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -245,11 +246,9 @@ public boolean perform(AbstractBuild<?, ?> build,
                         //git.push(null);
                     }
                 } catch (FormException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+					throw new AbortException("Failed to push merge to origin repository");
                 } catch (GitException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+					throw new AbortException("Failed to push merge to origin repository");
```

Comment:
Right :( Hope, something like ", because " + e.getMessage()" will be enough. Btw this should also remove stacktrace from build log and provide just an error.